sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorageTest.java

```diff
@@ -180,11 +194,14 @@ void record_ProcessesAttributes() {
     assertThat(logs.size()).isEqualTo(0);
   }
 
-  @Test
-  void record_MaxCardinality() {
+  @ParameterizedTest
+  @EnumSource(MemoryMode.class)
+  void record_MaxCardinality(MemoryMode memoryMode) {
+    setup(memoryMode);
```

Comment:
Is `@BeforeEach` not called before each iteration of a parameterized test?

Reply:
It does, but the parameter is injected as an argument to the `@ParameterizedTest`. I haven't clean way to inject the parameter into the `@BeforeEach` method. I saw JUnit is working on introducing parameter at Class level here https://github.com/junit-team/junit5/issues/878.